langchain4j-core/src/main/java/dev/langchain4j/model/chat/observability/ChatLanguageModelListener.java

```diff
@@ -0,0 +1,48 @@
+package dev.langchain4j.model.chat.observability;
+
+import dev.langchain4j.Experimental;
+
+/**
+ * TODO
+ */
+// TODO name: ChatLanguageModelObserver?
+// TODO package
+@Experimental
+public interface ChatLanguageModelListener {
+
+
+    /**
+     * TODO
+     *
+     * @param request
+     */
+    // TODO names
+    @Experimental
+    default void onRequest(ChatLanguageModelRequest request) {
+
+    }
+
+    /**
+     * TODO
+     *
+     * @param request
+     * @param response
+     */
+    // TODO names
+    @Experimental
+    default void onResponse(ChatLanguageModelRequest request, ChatLanguageModelResponse response) {
+
+    }
+
+    /**
+     * TODO
+     *
+     * @param request
+     * @param error
+     */
+    // TODO names
+    @Experimental
+    default void onError(ChatLanguageModelRequest request, Throwable error) {
```

Comment:
should this method be non-default to enforce the user to handle error?

Reply:
I don't think it has to